src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -116,8 +127,8 @@ public void collectOpenshiftMetrics(String account, OffsetDateTime start, Offset
                         BigDecimal time = measurement.get(0);
                         BigDecimal value = measurement.get(1);
 
-                        Event event = MeteringEventFactory.openShiftClusterCores(account, clusterId,
-                            sla, usage, clock.dateFromUnix(time), value.doubleValue());
+                        Event event = createOrUpdateEvent(existing, account, clusterId, sla, usage, time,
+                            value);
                         events.add(event);
                         eventCount++;
```

Comment:
Because this `eventCount` may be inaccurate when processing conflicting events (see `Map` suggestion), this var will need to be replaced by `events.size()`.

Reply:
DONE